errors/errors.go

```diff
@@ -150,28 +150,37 @@ func (be *BoulderError) WithSubErrors(subErrs []SubBoulderError) *BoulderError {
 	}
 }
 
-// New is a convenience function for creating a new BoulderError
-func New(errType ErrorType, msg string, args ...interface{}) error {
+// New is a convenience function for creating a new BoulderError.
+func New(errType ErrorType, msg string) error {
+	return &BoulderError{
+		Type:   errType,
+		Detail: msg,
+	}
+}
+
+// Newf is a convenience function for creating a new BoulderError with a
```

Comment:
Since Newf is internal-only, this could be unexported.

Though also I note that the new, constant `New()` is only used one place; that place could use `"%s", inErrMsg` instead.

Reply:
Good call on un-exporting Newf.

I'd like to leave `.New()` as the only public interface here, though -- we generally shouldn't be using these constructors outside of this package, since we have all the type-specific constructors instead. The only exception is the grpc/errors package, which should never be setting a message of its own.